src/test/java/seedu/edulink/logic/commands/FilterCommandTest.java

```diff
@@ -0,0 +1,142 @@
+import static seedu.edulink.testutil.TypicalPersons.BENSON;
+import static seedu.edulink.testutil.TypicalPersons.DANIEL;
+import static seedu.edulink.testutil.TypicalPersons.getTypicalAddressBook;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.edulink.model.Model;
+import seedu.edulink.model.ModelManager;
+import seedu.edulink.model.UserPrefs;
+import seedu.edulink.model.student.Student;
+import seedu.edulink.model.student.TagsContainQueryTagsPredicate;
+import seedu.edulink.model.tag.Tag;
+
+public class FilterCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void equals() {
+
+        Tag testTag1 = new Tag("TA");
+        Tag testTag2 = new Tag("Smart");
+        Set<Tag> queryTags1 = new HashSet<>();
+        queryTags1.add(testTag1);
+        queryTags1.add(testTag2);
+
+        Tag testTag3 = new Tag("Hardworking");
+        Set<Tag> queryTags2 = new HashSet<>();
+        queryTags2.add(testTag3);
+
+        Predicate<Student> firstPredicate =
+                new TagsContainQueryTagsPredicate(queryTags1);
+        Predicate<Student> secondPredicate =
+                new TagsContainQueryTagsPredicate(queryTags2);
+
+        FilterCommand filterFirstCommand = new FilterCommand(firstPredicate);
+        FilterCommand filterSecondCommand = new FilterCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(filterFirstCommand.equals(filterFirstCommand));
+
+        // same values -> returns true
+        FilterCommand filterFirstCommandCopy = new FilterCommand(firstPredicate);
+        assertTrue(filterFirstCommand.equals(filterFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(filterFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(filterFirstCommand.equals(null));
+
+        // different name -> returns false
+        assertFalse(filterFirstCommand.equals(filterSecondCommand));
+    }
+
+    @Test
+    public void execute_nonExistentTags_noPersonFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 0);
+
+        Tag testTag1 = new Tag("Intelligent");
+        Tag testTag2 = new Tag("Smart");
+        Set<Tag> queryTags = new HashSet<>();
+        queryTags.add(testTag1);
+        queryTags.add(testTag2);
+
+        TagsContainQueryTagsPredicate predicate = new TagsContainQueryTagsPredicate(queryTags);
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredPersonList());
```

Comment:
I like how you also take care of the filtered list instead of just checking models.